Packs/Grafana/Integrations/Grafana/Grafana.py

```diff
@@ -0,0 +1,842 @@
+        response = self._concatenate_urls(response, 'avatarUrl')
+
+        return response
+
+    def user_get_by_id_request(self, user_id: str):
+        response = self._http_request('GET', f'api/users/{user_id}', headers=self._headers)
+        response = self._concatenate_url(response, 'avatarUrl')
+
+        return response
+
+    def users_teams_request(self, user_id: str):
+        response = self._http_request('GET', f'api/users/{user_id}/teams', headers=self._headers)
+        response = self._concatenate_urls(response, 'avatarUrl')
+
+        return response
+
+    def users_organization_request(self, user_id: str):
+        response = self._http_request('GET', f'api/users/{user_id}/orgs', headers=self._headers)
+
+        return response
+
+    def user_update_request(self, user_id: str, email: str = None, name: str = None, login: str = None,
+                            theme: str = None):
+        data = {"email": email, "login": login, "name": name, "theme": theme}
+
+        response = self._http_request('PUT', f'api/users/{user_id}', json_data=data, headers=self._headers)
+
+        return response
+
+    def annotation_create_request(self, text: str, dashboard_id: int = None, panel_id: int = None,
+                                  time_start: int = None, time_end: int = None, tags: List[str] = None):
+        data = {"dashboardId": dashboard_id, "panelId": panel_id,
+                "tags": tags, "text": text, "time": time_start, "timeEnd": time_end}
+
+        response = self._http_request('POST', 'api/annotations', json_data=data, headers=self._headers)
+
+        return response
+
+    def teams_search_request(self, page_size: str = None, page_number: str = None, query: str = None, name: str = None):
+        params = assign_params(perpage=page_size, page=page_number, query=query, name=name)
+
+        response = self._http_request('GET', 'api/teams/search', params=params, headers=self._headers)
+        response = self._concatenate_urls(response['teams'], 'avatarUrl')
+
+        return response
+
+    def team_get_by_id_request(self, team_id: str):
+        response = self._http_request('GET', f'api/teams/{team_id}', headers=self._headers)
+        response = self._concatenate_url(response, 'avatarUrl')
+
+        return response
+
+    def team_members_request(self, team_id: str):
+        response = self._http_request('GET', f'api/teams/{team_id}/members', headers=self._headers)
+        response = self._concatenate_urls(response, 'avatarUrl')
+
+        return response
+
+    def user_add_to_team_request(self, team_id: str, user_id: str):
+        data = {"userId": int(user_id)}
+
+        # 400 - 'User is already added to this team' might be raised - which isn't a real error
+        response = self._http_request('POST', f'api/teams/{team_id}/members', json_data=data, headers=self._headers,
+                                      ok_codes=(200, 400))
+
```

Comment:
Maybe we should think about telling the client that the user is already added to the team when we get `400 code`.

Reply:
This is what happens - in addition to the `400 code`, Grafana returns a message saying `"Member added to Team"` or `"User is already added to this team"`, that we show to the user.